executor.go

```diff
@@ -110,6 +111,14 @@ type ExecutionContext struct {
 	VariableValues map[string]interface{}
 	Errors         []gqlerrors.FormattedError
 	Context        context.Context
+
+	errorsMutex sync.Mutex
+}
+
+func (eCtx *ExecutionContext) addError(err gqlerrors.FormattedError) {
+	eCtx.errorsMutex.Lock()
+	eCtx.Errors = append(eCtx.Errors, err)
+	eCtx.errorsMutex.Unlock()
```

Comment:
I'd recommend, even though in this case it's benefits are null, a `defer eCtx.errorsMutex.Unlock()` immediately after the lock. The benefit here is potential reduction in future errors if this gets modified. This is just a style comment though.